### PR TITLE
Adds feature of character selection in BuildOutputTreeCellView.cs

### PR DESCRIPTION
**Pull request by netonjm**
Includes character selection and copy/paste behaviour in OutputTree

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputTreeCellView.cs [hunk 1]
```diff
@@ -114,6 +114,16 @@ class ViewStatus
 		static Font defaultLightFont;
 		static Font defaultBoldFont;
 
+		// Used to track the selection
+		int selectionStart;
+		int selectionEnd;
+
+		public int SelectionStart => selectionStart;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputTreeCellView.cs]: Convert to auto-property:
`public int SelectionStart { get; private set; }`

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs [hunk 2]
```diff
@@ -339,7 +339,13 @@ void TreeView_ButtonPressed (object sender, ButtonEventArgs e)
 				menu.Add (new SeparatorContextMenuItem ());
 				var copyElementMenu = new ContextMenuItem (GettextCatalog.GetString ("Copy\t\t\t{0}", GetShortcut (EditCommands.Copy, false)));
 				copyElementMenu.Clicked += (s, args) => {
-					ClipboardCopy (selectedNode);
+					if (cellView.SelectionStart != cellView.SelectionEnd) {
+						var init = Math.Min (cellView.SelectionStart, cellView.SelectionEnd);
+						var end = Math.Max (cellView.SelectionStart, cellView.SelectionEnd);
+						Clipboard.SetText (selectedNode.Message.Substring (init, end - init));
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs]: Isn't SetText obsolete?

Comment (netonjm) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs]: I used the same behaviour like in ErrorListPad

**Comment by rodrmoya** [comment]
Selection seems to work ok, except on rows with very long text that spans over the tree view height, but copying doesn't work for me, it always copies the whole row's text, not the selected text

**Pull request closed without merging**